Library/Homebrew/extend/os/info.rb

```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
+
+require "extend/os/mac/info" if OS.mac? || ENV["HOMEBREW_USE_HOMEBREW_ON_LINUX"]
+require "extend/os/linux/info" if OS.linux? && !ENV["HOMEBREW_USE_HOMEBREW_ON_LINUX"]
```

Comment:
I'd suggest the `ENV["HOMEBREW_USE_HOMEBREW_ON_LINUX"]` checks happen within the `extend/os/linux/info.rb `